repos.go

```diff
@@ -440,6 +440,34 @@ type Commit struct {
 	} `json:"files"`
 }
 
+// Compare represents detail of comparison between two branches or commits
+type Compare struct {
+	URL             string   `json:"url"`
+	HTMLURL         string   `json:"html_url"`
+	PermalinkURL    string   `json:"permalink_url"`
+	DiffURL         string   `json:"diff_url"`
+	PatchURL        string   `json:"patch_url"`
+	BaseCommit      Commit   `json:"base_commit"`
+	MergeBaseCommit Commit   `json:"merge_base_commit"`
+	Status          string   `json:"status"`
+	AheadBy         int      `json:"ahead_by"`
+	BehindBy        int      `json:"behind_by"`
+	TotalCommits    int      `json:"total_commits"`
+	Commits         []Commit `json:"commits"`
+	Files           []struct {
+		Sha         string `json:"sha"`
```

Comment:
@adamcroissant should be `SHA` but this is a 0.x library so I'm fine breaking the API if we want to update later. I'm sure there are other examples which use `Sha`. I should ask Matt if he'd update json2go for this 😄